File: src/commands/install-agents.js
"use strict";

const fs = require("fs");
const path = require("path");
const { getAgentsDir, getIdeAgentsTarget, readConfig } = require("../config");

async function installAgentsCommand(cwd) {
  const { checkbox } = require("@inquirer/prompts");
  cwd = cwd || process.cwd();
  const config = readConfig(cwd);

  if (!config) {
    console.log('No wizard configuration found. Run "wizard install" first.');
    return;
  }

  const agentsDir = getAgentsDir();
  const agentFiles = fs
    .readdirSync(agentsDir)
    .filter((f) => f.endsWith(".md"));

  if (agentFiles.length === 0) {
    console.log("No agent templates available.");
    return;
  }

  const choices = agentFiles
    .sort()
    .map((f) => ({ name: f.replace(".md", ""), value: f }));

  const selected = await checkbox({
    message: "Select agents to install:",
    choices,
  });

  if (selected.length === 0) {
    console.log("No agents selected. Aborting.");
    return;
  }

  const targets = getIdeAgentsTarget(cwd, config.ides);

  for (const [ide, targetDir] of Object.entries(targets)) {
    fs.mkdirSync(targetDir, { recursive: true });
    for (const agent of selected) {
      const src = path.join(agentsDir, agent);
      const dest = path.join(targetDir, agent);
      fs.copyFileSync(src, dest);
    }
    console.log(`Agents installed to ${path.relative(cwd, targetDir)}/`);
  }
}

module.exports = { installAgentsCommand };

File: wizard/commands/install_agents.py
"""Install agents command."""

import os
import shutil

from InquirerPy import inquirer

from wizard.config import get_agents_dir, get_ide_agents_target, read_config


def install_agents_command(cwd: str | None = None) -> None:
    """Run the install agents command."""
    cwd = cwd or os.getcwd()
    config = read_config(cwd)

    if not config:
        print('No wizard configuration found. Run "wizard install" first.')
        return

    agents_dir = get_agents_dir()
    agent_files = [f for f in os.listdir(agents_dir) if f.endswith(".md")]

    if not agent_files:
        print("No agent templates available.")
        return

    choices = [{"name": f.replace(".md", ""), "value": f} for f in sorted(agent_files)]

    selected = inquirer.checkbox(
        message="Select agents to install:",
        choices=choices,
    ).execute()

    if not selected:
        print("No agents selected. Aborting.")
        return

    targets = get_ide_agents_target(cwd, config["ides"])

    for ide, target_dir in targets.items():
        os.makedirs(target_dir, exist_ok=True)
        for agent in selected:
            src = os.path.join(agents_dir, agent)
            dest = os.path.join(target_dir, agent)
            shutil.copy2(src, dest)
        print(f"Agents installed to {os.path.relpath(target_dir, cwd)}/")
